pkg/kubectl/describe.go

```diff
@@ -1882,6 +1886,70 @@ func (p *PetSetDescriber) Describe(namespace, name string, describerSettings Des
 	})
 }
 
+type CertificateSigningRequestDescriber struct {
+	client *client.Client
+}
+
+func (p *CertificateSigningRequestDescriber) Describe(namespace, name string, describerSettings DescriberSettings) (string, error) {
+	csr, err := p.client.Certificates().CertificateSigningRequests().Get(name)
+	if err != nil {
+		return "", err
+	}
+
+	cr, err := utilcertificates.ParseCertificateRequestObject(csr)
+	if err != nil {
+		return "", fmt.Errorf("Error parsing CSR: %v", err)
+	}
+
+	return tabbedString(func(out io.Writer) error {
+		fmt.Fprintf(out, "Name:\t%s\n", csr.Name)
+		fmt.Fprintf(out, "Labels:\t%s\n", labels.FormatLabels(csr.Labels))
+		fmt.Fprintf(out, "Annotations:\t%s\n", labels.FormatLabels(csr.Annotations))
+		fmt.Fprintf(out, "CreationTimestamp:\t%s\n", csr.CreationTimestamp.Time.Format(time.RFC1123Z))
+		fmt.Fprintf(out, "RequestingUser:\t%s\n", csr.Spec.Username)
+
+		fmt.Fprintf(out, "Subject:\n")
+		fmt.Fprintf(out, "\tCommonName:\t%s\n", cr.Subject.CommonName)
+		fmt.Fprintf(out, "\tSerialNumber:\t%s\n", cr.Subject.SerialNumber)
+		fmt.Fprintf(out, "\tOrganization:\t%s\n", strings.Join(cr.Subject.Organization, ","))
+		fmt.Fprintf(out, "\tOrganizationalUnit:\t%s\n", strings.Join(cr.Subject.OrganizationalUnit, ","))
+		fmt.Fprintf(out, "\tCountry:\t%s\n", strings.Join(cr.Subject.Country, ","))
+		fmt.Fprintf(out, "\tLocality:\t%s\n", strings.Join(cr.Subject.Locality, ","))
+		fmt.Fprintf(out, "\tProvince:\t%s\n", strings.Join(cr.Subject.Province, ","))
+		fmt.Fprintf(out, "\tStreetAddress:\t%s\n", strings.Join(cr.Subject.StreetAddress, ","))
+		fmt.Fprintf(out, "\tPostalCode:\t%s\n", strings.Join(cr.Subject.PostalCode, ","))
```

Comment:
for the multi-value ones, can we print additional values on newlines to be able to distinguish between ["foo,bar"] and ["foo","bar"]? For example:
```
  Organization: foo
                bar
```
empty and single-value lists (the 99% case, mostly likely) wouldn't take up any more room